Automattic/pocket-casts-android
# Fix: Search Term Persists When Navigating to Different Podcast Page

## Description
Fixes the wrong episode search persistence 

Fixes #1320 

## Testing Instructions
1. Open an podcast
2. Search something
3. Go to another view, such as another podcast
4. You should see the search term removed

## Screenshots or Screencast 
[Screen_recording_20240529_091351.webm](https://github.com/Automattic/pocket-casts-android/assets/42220351/2d437476-1189-4ddc-80d2-d64de0973396)


## Checklist
 
#### I have tested any UI changes...
